verifier/verifier.go

```diff
@@ -65,13 +65,35 @@ type VerificationResult struct {
 	// Never is a list of certificate chains that could never be valid due to
 	// date-related issues, but are otherwise valid.
 	Never [][]*x509.Certificate
+
+	// CertificateType is one of Leaf, Intermediate, or Root.
+	CertificateType x509.CertificateType
+
+	// VerifyTime is time used in verification, set in the VerificationOptions.
+	VerifyTime time.Time
+
+	// InValidityWindow is true if NotBefore < VerifyTime < NotAfter
+	InValidityWindow bool
 }
 
 // MatchesDomain returns true if NameError == nil and Name != "".
 func (res *VerificationResult) MatchesDomain() bool {
 	return res.NameError == nil && res.Name != ""
 }
 
+// HasTrustedChain returns true if len(current) > 0
+func (res *VerificationResult) HasTrustedChain() bool {
+	return len(res.Current) > 0
+}
+
+// HadTrustedChain returns true if at some point in time, the certificate had a
+// chain to a trusted root in this store.
+//
+// This is equivalent to checking if len(current) > 0 || len(expired) > 0
+func (res *VerificationResult) HadTrustedChain() bool {
+	return res.HasTrustedChain() || len(res.Expired) > 0
```

Comment:
I would have expected Expired to be a boolean that described whether the certificate was expired. Perhaps we should find a more descriptive term for this field.